### PR TITLE
Move theming support from Beta to Main

### PR DESCRIPTION
**Pull request by jordanjones243**
# Alaska Airlines Pull Request

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

**Resolves:** #30 

## Summary:

_Please summarize the scope of the changes you have submitted, what the intent of the work is and anything that describes the before/after state of the project._

## Type of change:

Please delete options that are not relevant.

- [x] New capability
- [x] Revision of an existing capability
- [ ] Infrastructure change (automation, etc.)
- [ ] Other (please elaborate)

## Checklist:

- [x] My update follows the CONTRIBUTING guidelines of this project
- [x] I have performed a self-review of my own update

**By submitting this Pull Request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

_Pull Requests will be evaluated by their quality of update and whether it is consistent with the goals and values of this project. Any submission is to be considered a conversation between the submitter and the maintainers of this project and may require changes to your submission._

**Thank you for your submission!**<br>
-- Auro Design System Team

## Summary by Sourcery

Move theming support from Beta to Main by implementing color theming capabilities for the Auro Sidenav component, refactoring the component for better version management, and updating documentation to reflect these changes. Enhance code maintainability by restructuring styles and adding an ESLint configuration.

New Features:
- Introduce color theming support for the Auro Sidenav component, allowing customization of colors through design tokens.

Enhancements:
- Refactor the Auro Sidenav component to use dynamic accordion tags for better version management.
- Improve the styling structure by separating color styles into dedicated SCSS files for better maintainability.

Documentation:
- Update documentation to include information on the new theming capabilities and how to use design tokens for customization.

Tests:
- Modify tests to ensure compatibility with the new accordion tag implementation.

Chores:
- Add an ESLint configuration file to standardize code quality checks across the project.

### CODE REVIEW

File: test/auro-sidenav.test.js [hunk 1]
```diff
@@ -103,7 +103,7 @@ describe('auro-sidenav', () => {
       <auro-sidenav></auro-sidenav>
     `);
 
-    expect(el.shadowRoot.querySelector('auro-accordion')).to.not.be.null;
+    expect(el.shadowRoot.getElementById('accordion')).to.not.be.null;
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply test/auro-sidenav.test.js:106]: Not updating test at this time.

File: demo/api.md [hunk 2]
```diff
@@ -120,4 +120,28 @@ The `heading` slot defines what to use as the header of the sidenav.
 </auro-sidenav>
 ```
 <!-- AURO-GENERATED-CONTENT:END -->
-</auro-accordion>
+</auro-accordion>
+
+### Theme Support
+
+The component may be restyled using the following code sample and changing the values of the following token(s).
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply demo/api.md:128]: Not updating verbiage at this time

File: src/auro-sidenav.js [hunk 3]
```diff
@@ -29,6 +31,17 @@ import styleCss from "./style-css.js";
...
 export class AuroSidenav extends LitElement {
+  constructor() {
+    super();
+
+    const versioning = new AuroDependencyVersioning();
+
+    /**
+     * @private
+     */
+    this.accordionTag = versioning.generateTag('auro-accordion', accordionVersion, AuroAccordion);
```
[the remark this answers is not in the document]
Comment (jordanjones243) [reply src/auro-sidenav.js:44]: Not editing property at this time.

**Review by jason-capsule42 — APPROVED** [review]

**Pull request merged**